packages/components/textarea/src/textarea.spec.ts

```diff
@@ -0,0 +1,351 @@
+
+      expect(el).to.have.attribute('readonly');
+    });
+
+    it('should not have an autocomplete by default', () => {
+      expect(el).not.to.have.attribute('autocomplete');
+    });
+
+    it('should not have an autocomplete when set', async() => {
+      el.autocomplete = 'off';
+      await el.updateComplete;
+
+      const textarea = el.querySelector('textarea');
+
+      expect(textarea).to.have.attribute('autocomplete', 'off');
+    });
+
+    it('should not have a placeholder by default', () => {
+      expect(el).not.to.have.attribute('placeholder');
+    });
+
+    it('should have a placeholder when set', async () => {
+      el.placeholder = 'my placeholder';
+      await el.updateComplete;
+
+      const textarea = el.querySelector('textarea');
+
+      expect(textarea).to.have.attribute('placeholder', 'my placeholder');
+    });
+
+    it('should not have a minLength by default', () => {
+      expect(el).not.to.have.attribute('minLength');
+    });
+
+    it('should not have a maxLength by default', () => {
+      expect(el).not.to.have.attribute('maxLength');
+    });
+
+    it('should not be invalid by default', () => {
+      expect(el).not.to.have.attribute('invalid');
+    });
+
+    it('should not be valid by default', () => {
+      expect(el).not.to.have.attribute('valid');
+    });
+
+    it('should not have showValid by default', () => {
+      expect(el).not.to.have.attribute('showValid');
+    });
+
+    it('should not have a autocomplete attribute when autocomplete property is not provided', async() => {
+      el.autocomplete = "off";
+      await el.updateComplete;
+
+      el.autocomplete = undefined;
+      await el.updateComplete;
+
+      expect(el.textarea.getAttribute('autocomplete')).to.be.null;
+    });
+
+    it('should have a resize attribute of vertical when resize property is not provided', async() => {
+      el.resize = "auto";
+      await el.updateComplete;
+
+      (el.resize as ResizeType | undefined) = undefined;
+      await el.updateComplete;
+
+      expect(el.textarea.getAttribute('resize')).to.equal('vertical');
+    });
+
+    it('should have a wrap attribute of soft value when wrap property is not provided', async() => {
+      el.wrap = "hard";
+      await el.updateComplete;
+
+      (el.wrap as WrapType | undefined) = undefined;
+      await el.updateComplete;
+
+      expect(el.textarea.getAttribute('wrap')).to.equal('soft');
+    });
+
+    it('should not have maxlength attribute when maxLength property is not provided', async() => {
+      el.maxLength = 8;
+      await el.updateComplete;
+
+      el.maxLength = undefined;
+      await el.updateComplete;
+
+      expect(el.textarea.getAttribute('maxlength')).to.be.null;
+    });
+
+    it('should not have minlength attribute when minLength property is not provided', async() => {
+      el.minLength = 2;
+      await el.updateComplete;
+
+      el.minLength = undefined;
+      await el.updateComplete;
```

Comment:
why set it to `2` and then to undefined without checking if the `2` is being set properly?